pkg/client/uri.go

```diff
@@ -42,12 +44,26 @@ func (as *UriService) Create(ctx context.Context, input CreateUriInput) (*Create
 func (as *UriService) Read(ctx context.Context, id string) (*ReadUriResult, error) {
 	log.Printf("read uri request. id=%s", id)
 
-	resp, err := getUriById(ctx, as.client.gql, id)
+	operation := func() (getUriByIdEntitiesEntityQueriesByIdEntity, error) {
+		resp, err := getUriById(ctx, as.client.gql, id)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.Entities.ById == nil {
+			return nil, backoff.RetryAfter(1)
```

Comment:
Ah.

`backoff.RetryAfter(1)` returns an error that causes Retry to reset its backoff timer.

https://github.com/cenkalti/backoff/blob/v5/retry.go#L114-L119

That's not what we want to happen.

We should consider the following:

If the error returned is caused by a 404, we want to retry--so we can return some basic error that will cause a retry... we could even just return the error provided by `getUriById` just to keep it simple. That means that the error from `getUriById` must return some identifiable error if there is a 404, meaning the client will have to let us know if there was an error that was not found in the API.

If we get a non-404 error, then we need to throw a [PermanentError](https://github.com/cenkalti/backoff/blob/v5/error.go#L8-L21) so that we don't retry. We could get a little fancy and retry on temporary failures, but let's get the "not found" case handled first.